crates/bevy_ecs/src/schedule_v3/state.rs

```diff
@@ -0,0 +1,60 @@
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::mem;
+
+use crate as bevy_ecs;
+use crate::schedule_v3::{ScheduleLabel, SystemSet, WorldExt};
+use crate::system::Resource;
+use crate::world::World;
+
+/// Types that can define states in a finite-state machine.
+pub trait Statelike: 'static + Send + Sync + Clone + PartialEq + Eq + Hash + Debug {}
```

Comment:
This is currently missing the add_state functionality we outlined in the RFC (and discussed here: https://github.com/bevyengine/rfcs/pull/45#discussion_r953218580)

Reply:
Yep, this should be a method on `App`.

Reply:
Agreed!